bulk_update() fails if PK is UUID

**Describe the bug**
Running bulk_update on a model which has UUID as its Primary Key causes a Postgres syntax error; I suspect because the UUID is not quoted in the query.


**To Reproduce**
ExampleModel:
```
class ExampleModel(Model):
    id = fields.UUIDField(pk=True, default=uuid4)
    is_submitted = fields.BooleanField(default=False)
```

```
em1 = await ExampleModel.get(id="1f5a40d8-27c4-437f-957e-c0bd85bc6795")
em1.is_submitted = True

em2 = await ExampleModel.get(id="3f10348e-2535-4e1f-af38-429b85262fa5")
em2.is_submitted = True

await ExampleModel.bulk_update([em1, em2], fields=["is_submitted"])
```

**Expected behavior**
Two rows should have their `is_submitted` values updated

**Additional context**
tortoise-orm 0.17.8
python 3.9.1

Console output:
```
self = <tortoise.backends.asyncpg.client.AsyncpgDBClient object at 0x7f8d028b12e0>
args = ('UPDATE "ExampleModel" SET "is_submitted"=CASE WHEN "id"=1f5a40d8-27c4-437f-957e-c0bd85bc6795 THEN true WHEN "id"=3f10348e-2535-4e1f-af38-429b85262fa5 THEN true END WHERE "id" IN (1f5a40d8-27c4-437f-957e-c0bd85bc6795,3f10348e-2535-4e1f-af38-429b85262fa5)',)

    @wraps(func)
    async def translate_exceptions_(self, *args):
        try:
            return await func(self, *args)
        except (asyncpg.SyntaxOrAccessError, asyncpg.exceptions.DataError) as exc:
>           raise OperationalError(exc)
E           tortoise.exceptions.OperationalError: syntax error at or near "f5a40d"

/usr/local/lib/python3.9/site-packages/tortoise/backends/asyncpg/client.py:38: OperationalError
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -14,6 +14,9 @@ Added
 ^^^^^
 - Add Case-When support. (#943)
 - Add `Rand`/`Random` function in contrib. (#944)
+Fixed
+^^^^^
+- Fix `bulk_update` error when pk is uuid. (#986)
 Changed
 ^^^^^^^
 - Move `Function`, `Aggregate` from `functions.py` to `expressions.py`. (#943)

--- a/tests/test_update.py
+++ b/tests/test_update.py
@@ -1,9 +1,10 @@
+import uuid
 from datetime import datetime, timedelta
 from typing import Any
 
 from pypika.terms import Function
 
-from tests.testmodels import DefaultUpdate, Event, IntFields, JSONFields, Tournament
+from tests.testmodels import DefaultUpdate, Event, IntFields, JSONFields, Tournament, UUIDFields
 from tortoise.contrib import test
 from tortoise.expressions import F
 
@@ -26,6 +27,18 @@ class TestUpdate(test.TestCase):
         self.assertEqual(rows_affected, 2)
         self.assertEqual((await Tournament.get(pk=objs[0].pk)).name, "0")
         self.assertEqual((await Tournament.get(pk=objs[1].pk)).name, "1")
+
+    async def test_bulk_update_pk_uuid(self):
+        objs = [
+            await UUIDFields.create(data=uuid.uuid4()),
+            await UUIDFields.create(data=uuid.uuid4()),
+        ]
+        objs[0].data = str(uuid.uuid4())
+        objs[1].data = str(uuid.uuid4())
+        rows_affected = await UUIDFields.bulk_update(objs, fields=["data"])
+        self.assertEqual(rows_affected, 2)
+        self.assertEqual((await UUIDFields.get(pk=objs[0].pk)).data, objs[0].data)
+        self.assertEqual((await UUIDFields.get(pk=objs[1].pk)).data, objs[1].data)
 
     async def test_update_auto_now(self):
         obj = await DefaultUpdate.create()

--- a/tortoise/queryset.py
+++ b/tortoise/queryset.py
@@ -33,7 +33,7 @@ from tortoise.exceptions import (
     MultipleObjectsReturned,
     ParamsError,
 )
-from tortoise.expressions import F, Q, RawSQL
+from tortoise.expressions import Expression, F, Q, RawSQL
 from tortoise.fields.relational import (
     ForeignKeyFieldInstance,
     OneToOneFieldInstance,
@@ -92,7 +92,7 @@ class AwaitableQuery(Generic[MODEL]):
         self.query: QueryBuilder = QUERY
         self._db: BaseDBAsyncClient = None  # type: ignore
         self.capabilities: Capabilities = model._meta.db.capabilities
-        self._annotations: Dict[str, Function] = {}
+        self._annotations: Dict[str, Expression] = {}
 
     def _choose_db(self, for_write: bool = False) -> BaseDBAsyncClient:
         """
@@ -474,7 +474,7 @@ class QuerySet(AwaitableQuery[MODEL]):
             return queryset
         return self
 
-    def annotate(self, **kwargs: Union[Function, Term]) -> "QuerySet[MODEL]":
+    def annotate(self, **kwargs: Union[Expression, Term]) -> "QuerySet[MODEL]":
         """
         Annotate result with aggregation or function result.
 
@@ -1649,16 +1649,19 @@ class BulkUpdateQuery(UpdateQuery):
             annotations=self.annotations,
             custom_filters=self.custom_filters,
         )
-        pk = Field(self.model._meta.pk_attr)
+        executor = self._db.executor_class(model=self.model, db=self._db)
+        pk_attr = self.model._meta.pk_attr
+        pk = Field(pk_attr)
         for objects_item in chunk(self.objects, self.batch_size):
             query = copy(self.query)
             for field in self.fields:
                 case = Case()
                 pk_list = []
                 for obj in objects_item:
+                    value = executor.column_map[pk_attr](obj.pk, None)
                     attr = getattr(obj, field)
-                    case.when(pk == obj.pk, attr)
-                    pk_list.append(obj.pk)
+                    case.when(pk == value, attr)
+                    pk_list.append(value)
                 query = query.set(field, case)
                 query = query.where(pk.isin(pk_list))
             self.queries.append(query)

